src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/CompilationContext.cs

```diff
@@ -1230,6 +1240,8 @@ private static DkmClrCompilationResultFlags GetLocalResultFlags(LocalSymbol loca
                 var fieldType = field.Type;
                 var fieldKind = GeneratedNames.GetKind(field.Name);
                 if (fieldKind == GeneratedNameKind.DisplayClassLocalOrField ||
+                    fieldKind == GeneratedNameKind.TransparentIdentifier ||
+                    (fieldKind == GeneratedNameKind.AnonymousTypeField && IsTransparentIdentifierFieldInAnonymousType(field)) ||
```

Comment:
It would be slightly less efficient, but consider moving "fieldKind == GeneratedNameKind.AnonymousTypeField" check inside IsTransparentIdentifierFieldInAnonymousType (otherwise, it's not necessarily "in anonymous type").